CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodeFile.swift

```diff
@@ -0,0 +1,359 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+/// This gyb-file generates the syntax nodes for SwiftSyntax. To keep the generated
+/// files at a managable file size, it is to be invoked multiple times with the
+/// global variable EMIT_KIND set to a base kind listed in
+/// gyb_syntax_support/kinds.py:SYNTAX_BASE_KINDES. It then only emits those syntax
+/// nodes whose base kind are that specified kind.
+func syntaxNode(emitKind: String) -> SourceFileSyntax {
+  SourceFileSyntax(leadingTrivia: .docLineComment(generateCopyrightHeader(for: "generate-swiftsyntax"))) {
+    for node in SYNTAX_NODES {
+      if node.isBase {
+        // Handled in SyntaxNodesBase.swift.gyb
```

Comment:
Nit: `SyntaxNodesBase.swift.gyb` -> `SyntaxNodesBaseFile.swift`

(similar below).

Also, we could reduce nesting level by 1 by having `continue` in these branches and then the `else` block could come after this `if` statement.

Reply:
We cannot use `continue` due to `Closure containing control flow statement cannot be used with result builder 'CodeBlockItemListBuilder'`

Reply:
We could do the following instead and merge all the comments in the `if` branches into a single one above the loop.

```swift
    for node in SYNTAX_NODES where !node.isBase && node.collectionElement.isEmpty && node.baseKind == emitKind {
```